### PR TITLE
add operation for enabling automatic sidecar injection

### LINKED ISSUE #33: [configuration] add operation for enabling automatic sidecar injection
**Current Behavior**
There is no operation to prepare targeted namespace and enable nginx-sm automatic sidecar injection on it.

**Desired Behavior**
* An operation which would allow users to annotate a namespace and thus prepare it for auto sidecar injection.
* Whenever a sample app is deployed, this operation should be triggered automatically to ensure that the pods have NGINX sidecars injected.

**Resources**
For reference, see:
https://github.com/meshery/meshery-istio/blob/7d73a272e4886588bb3552018f0eae6f440aa2ea/istio/istio.go#L126
https://github.com/meshery/meshery-istio/blob/40d1ab790a50fd74f603c450da3d05984f8564db/istio/sample_apps.go#L84
https://github.com/meshery/meshery-istio/blob/40d1ab790a50fd74f603c450da3d05984f8564db/istio/sample_apps.go#L108

### Contributor Resources

The meshery.io website uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/meshery/meshery.io/tree/master).
- See the [Contributing to Meshery.io Website](https://github.com/layer5io/meshery.io#contributing-to-the-mesheryio-website) section of the readme.md and other [contributing instructions](https://docs.meshery.io/project/contributing), too.
- See Meshery site designs in this [Figma designfile](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI?node-id=110%3A1). Fill-in the [Layer5 Community Member Form](https://layer5.io/newcomer) and join the [Layer5 Community Slack](http://slack.layer5.io) for access.

### PR DESCRIPTION
**Pull request by Shreyas220**
Signed-off-by: Shreyas220 <shreyas.ny@gmail.com>

**Description**
- added operation to prepare targeted namespace and enable nginx-sm automatic sidecar injection on it
- fixes an instance of misused meshkit logger

This PR fixes #33

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by utkarsh-pro — APPROVED** [review]
LGTM

**Comment by leecalcote** [comment]
@Shreyas220 will you have a look at the merge conflict?

File: nginx/error.go [hunk 1]
```diff
@@ -68,3 +72,9 @@ func ErrCustomOperation(err error) error {
 func ErrApplyHelmChart(err error) error {
 	return errors.New(ErrApplyHelmChartCode, errors.Alert, []string{"Error occured while applying Helm Chart"}, []string{err.Error()}, []string{}, []string{})
 }
+
+// ErrLoadNamespace is the occurend while applying namespace
+func ErrLoadNamespace(err error, s string ) error{
+	return errors.New(ErrLoadNamespaceCode, errors.Alert, []string{"Error occured while applying namespace "}, []string{err.Error()}, []string{}, []string{})
```
Comment (leecalcote) [inline nginx/error.go:78]: Please include probable cause and suggested remediation on all errors.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: nginx/error.go [hunk 2]
```diff
@@ -99,3 +103,9 @@ func ErrNginxCoreComponentFail(err error) error {
 func ErrParseNginxCoreComponent(err error) error {
 	return errors.New(ErrParseNginxCoreComponentCode, errors.Alert, []string{"Failure to parse core component manifest for NGINX Service Mesh"}, []string{err.Error()}, []string{}, []string{})
 }
+
+// ErrLoadNamespace is the occurend while applying namespace
+func ErrLoadNamespace(err error, s string ) error{
+	return errors.New(ErrLoadNamespaceCode, errors.Alert, []string{"Error occured while applying namespace "}, []string{err.Error()}, []string{"Trying to access a namespace which is not available"}, []string{})
```
Comment (leecalcote) [inline nginx/error.go:109]: @Shreyas220 Suggested remediation needed. Could be: "Verify presence of namespace. Confirm Meshery ServiceAccount permissions."

**Comment by leecalcote** [comment]
@Revolyssup what do you think?

**Comment by Revolyssup** [comment]
> @Revolyssup what do you think?

gtg

**Review by utkarsh-pro — APPROVED** [review]
LGTM

**Pull request merged**